### PR TITLE
JATS: Fixed edge case handing for collaborations with nested authors

### PR DESCRIPTION
**Pull request by mugdhapolimera**

### CODE REVIEW

File: tests/stubdata/output/jats_springer_JHEP_JHEP07_2023_200.json [hunk 1]
```diff
@@ -11,6 +11,26 @@
         "collab": "The ALICE collaboration"
       }
     },
+    {
```
Comment (mugdhapolimera) [inline tests/stubdata/output/jats_springer_JHEP_JHEP07_2023_200.json:14]: This new change is because the collab is now treated as an "author" so all the affIDs are being parsed with one of the recent changes we made

**Review by seasidesparrow — APPROVED** [review]
Nice, looks good!
